core/proto/keytransparency_v1_types/keytransparency_v1_types.proto

```diff
@@ -251,3 +251,26 @@ message GetDomainInfoResponse {
   // Vrf contains the VRF public key.
   keyspb.PublicKey vrf = 3;
 }
+
+// SetEntriesRequest will update the accounts of the given user_ids to be set to
+// user data if those accounts have key_id in their set of authorized_keys.
+// No change to the authorized key set will be made.
+message BatchSetEntriesRequest {
+  // users is a map from user_ids to user data.
+  map<string,bytes> users = 1;
+
+  // app_id is the application to make this change for
+  string app_id = 2;
+
+  // key_id is the id of the authorized_public key to use when updating accounts.
+  // This must be a key that this server has the private key for.
+  string key_id = 3;
+}
+
+
+// BatchSetEntriesResponse returns a list of users for which the set operation
+// was unsuccessful.
+message BatchSetEntriesResponse {
+  // errors is a map from user_ids to errors, if there was an error for that user.
+  map<string, string> errors = 1;
```

Comment:
Yes, in it's current form the caller of the batch API trusts it to do the right thing.
Since this is a backend integration point, I thought this would be a reasonable trust model.